portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,51 +33,59 @@
 
 import java.util.List;
 import java.util.ArrayList;
-import java.util.Arrays;
-
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    private static List<String> comments = new ArrayList<>();
-
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // List<String> list = new ArrayList<>(Arrays.asList(new String[]{"First message", "Second message", "Third message"}));
-    String json = convertGson(comments);
-
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
-  }
-
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String comment = getComment(request.getParameter("comment-text"));
-    long timestamp = System.currentTimeMillis();
-
-    Entity newComment = new Entity("Comment");
-    newComment.setProperty("text", comment);
-    newComment.setProperty("time", timestamp);
-
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(newComment);
-
-    response.sendRedirect("/index.html");
-  }
-
-  private String getComment(HttpServletRequest request) {
-      String commentString = request.getParameter("comment-text");
-
-      if (commentString.length() > 0) {
-          return commentString;
-      } else {
-          return "They thought...";
-      }
-  }
-
-  private String convertGson(List<String> element) {
-      Gson gson = new Gson();
-      String json = gson.toJson(element);
-      return json;
-  }
+    private Integer numComments = Comment.getNumComments();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Comment> comments = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId();
+        String text = (String) entity.getProperty("text");
+        long timestamp = (long) entity.getProperty("time");
+
+        Comment comment = new Comment(id, text, timestamp);
+        comments.add(comment);
+    }
+
+        Gson gson = new Gson();
+
+        String json = "{";
+        json += "\"numComments\": \"" + numComments.toString() + "\", ";
+        json += "\"commentList\":" + gson.toJson(comments) + "}";
+
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Please make sure this adheres to the Java style guide. It looks like this line exceeds 100 columns.

Reply:
I looked at the line and it was 102 lines, but when I ran the java formatting tool that you sent us it did not tell me to break it up onto two different lines.

Reply:
*102 columns